src/systems/filecoin_vm/runtime/runtime.id

```diff
@@ -24,10 +25,11 @@ type Runtime interface {
     ValidateImmediateCallerAcceptAny()
     ValidateImmediateCallerMatches(CallerPattern)
 
-    // The address of the actor receiving the message.
+    // The address of the actor receiving the message. Always an ID-address.
     CurrReceiver()         addr.Address
 
     // The actor who mined the block in which the initial on-chain message appears.
+    // Always an ID-address.
     ToplevelBlockWinner()  addr.Address
```

Comment:
how would you verify message sigs if passing in the id addr here? Lookup of miner key based on id addr each time..?

Reply:
seems convoluted. A good rule might be sender --> use key addr, receiver use id addr...

Reply:
I think actors should be able to treat addresses as black-boxes for the purposes of signature verification, i.e., shouldn't rely on being able to inspect them manually to obtain the public key. Rather they should be able to invoke a runtime method to verify a signature given an actor address, or (if we're willing to designate AccountActor code as trusted) by sending a message to the actor to verify its own signature. The former seems more intuitive to me, but I don't have a strong preference.

Reply:
Signature verification is always off-chain. The node needs to look up the pubkey-address from account actor state (which is why the account actor has state at all).